sdk/monitor/azquery/utils_test.go

```diff
@@ -103,18 +122,39 @@ func startLogsTest(t *testing.T) *azquery.LogsClient {
 	startRecording(t)
 	transport, err := recording.NewRecordingHTTPClient(t, nil)
 	require.NoError(t, err)
-	opts := &azquery.LogsClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport}}
+	opts := &azquery.LogsClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport, Cloud: clientCloud}}
 	return azquery.NewLogsClient(credential, opts)
 }
 
 func startMetricsTest(t *testing.T) *azquery.MetricsClient {
-	startRecording(t)
-	transport, err := recording.NewRecordingHTTPClient(t, nil)
-	require.NoError(t, err)
-	opts := &azquery.MetricsClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport}}
+	var opts *azquery.MetricsClientOptions
+	if recording.GetRecordMode() == recording.LiveMode {
+		transport := &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					Renegotiation: tls.RenegotiateOnceAsClient,
+				},
+			},
+		}
+		opts = &azquery.MetricsClientOptions{ClientOptions: azcore.ClientOptions{Transport: transport, Cloud: clientCloud}}
```

Comment:
Shouldn't we set the cloud in playback as well?

Reply:
We could set it here, but it's not necessary. The cloud during testing is only changed from the default public setting in live mode (lines 78-85). The tests automatically fail in playback mode with different clouds. It can't identify the recording since the URLs are different.